### PR TITLE
docs: typo cleanups for docs

### PR DESCRIPTION
**Pull request by nschonni**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x ] Other, please explain:

#### What changes did you make? (Give an overview)

- Fix some minor typos in the JSDocs
- Fix typos/missing hyphens in some MD docs
- Correct bad CSS value/typo
- Correct UK spelling of "behaviour" that is shown as US spelling "behavior" in intellisense.

#### Is there anything you'd like reviewers to focus on?

These could be split out to separate PRs, but I just kept the commits separate for now.

### CODE REVIEW

File: docs/src/assets/js/search.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ document.addEventListener('keydown', function (e) {
     if ((e.metaKey || e.ctrlKey) && e.key === 'k') {
         e.preventDefault();
         searchInput.focus();
-        document.querySelector('.search').scrollIntoView({ behaviour: "smooth", block: "start" });
+        document.querySelector('.search').scrollIntoView({ behavior: "smooth", block: "start" });
```
Comment (snitin315) [inline docs/src/assets/js/search.js:167]: Yeah, [it should be behavior](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView#examples)

**Review by snitin315 — APPROVED** [review]
LGTM. Changing the type to `docs` since there is a user-facing change.

File: docs/src/assets/scss/components/search.scss [hunk 2]
```diff
@@ -70,7 +70,7 @@
     width: 100%;
     border-radius: 0 0 var(--border-radius) var(--border-radius);
     border: 1px solid var(--divider-color);
-    position: rekative;
+    position: relative;
```
Comment (nschonni) [inline docs/src/assets/scss/components/search.scss:73]: Was going to try running stylelint to see if there is any other odd ones later

Comment (snitin315) [reply docs/src/assets/scss/components/search.scss:73]: We have a similar PR open https://github.com/eslint/eslint/pull/16226

**Comment by nschonni** [comment]
I've got a local script to regerate the `project-files.txt`, but you can see the local config I used here https://gist.github.com/nschonni/265577f3d51e1b1eea57e4da76263819. It's not really cleaned up, but I could PR something if you want

**Comment by nschonni** [comment]
This is my local git alias I use `spell = "!git add project-words.txt && cspell --no-progress --words-only --unique \"**\" > project-words.txt && sort -o project-words.txt project-words.txt"`, although the sort call doesn't work correctly. I just then scan down the list to see the issues, or on repos I've done this, use the diff view to quickly scan just the newly added words in the text file after sorting it in VSCode

**Pull request merged**